test/integration/smoke/test_persistent_network.py

```diff
@@ -52,6 +52,22 @@ def setUpClass(cls):
         cls.testClient = super(TestL2PersistentNetworks, cls).getClsTestClient()
         cls.api_client = cls.testClient.getApiClient()
         cls.hypervisor = cls.testClient.getHypervisorInfo()
+
+        isKVM = cls.hypervisor.lower() in ["kvm"]
+        isOVSEnabled = False
+        hostConfig = cls.config.__dict__["zones"][0].__dict__["pods"][0].__dict__["clusters"][0].__dict__["hosts"][0].__dict__
+        if isKVM :
+            # Test only if all the hosts use OVS
+            grepCmd = 'grep "network.bridge.type=openvswitch" /etc/cloudstack/agent/agent.properties'
+            hosts = list_hosts(cls.api_client, type='Routing', hypervisor='kvm')
+            if len(hosts) > 0 :
+                isOVSEnabled = True
+            for host in hosts :
+                isOVSEnabled = isOVSEnabled or len(SshClient(host.ipaddress, port=22, user=hostConfig["username"],
```

Comment:
If we want to ensure all hosts use OVS I think we need to add a check on the for loop and exit if `!isOVSEnabled`. Otherwise, we will only decide based on last host in the iteration

Reply:
Ignore that, I see that `isOvsEnabled` is set to true if there are hosts. New question: why do we even check for the second part of the OR if the first part is always true? The for loop may not be needed